src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -775,6 +777,30 @@ private static void addIntelliJRunConfiguration(Project project,
         );
         appRun.setMainClass(RunUtils.DEV_LAUNCH_MAIN_CLASS);
         appRun.setProgramParameters(RunUtils.escapeJvmArg(RunUtils.getArgFileParameter(prepareTask.getProgramArgsFile().get())));
+
+        if (!run.getTasksBefore().isEmpty()) {
+            // This is slightly annoying.
+            // idea-ext does not expose the ability to run multiple gradle tasks at once, but the IDE model is capable of it.
+            class GradleTasks extends BeforeRunTask {
+                @Inject
+                GradleTasks(String nameParam) {
+                    type = "gradleTask";
+                    name = nameParam;
+                }
+
+                @SuppressWarnings("unchecked")
+                @Override
+                public Map<String, ?> toMap() {
+                    var result = (Map<String, Object>) super.toMap();
+                    result.put("projectPath", project.getProjectDir().getAbsolutePath().replaceAll("\\\\", "/"));
+                    var tasks = run.getTasksBefore().stream().map(TaskProvider::getName).collect(Collectors.joining(" "));
```

Comment:
I don't think `getName`, you're interested in the full path of the run (including the project reference)